pkg/proxy/conntrack/conntrack.go

```diff
@@ -49,95 +50,141 @@ type Interface interface {
+func newConntracker(handler netlinkHandler) Interface {
+	return &conntracker{handler: handler}
+}
 
 func protoStr(proto v1.Protocol) string {
 	return strings.ToLower(string(proto))
 }
 
-func parametersWithFamily(isIPv6 bool, parameters ...string) []string {
+// getNetlinkFamily returns the Netlink IP family constant
+func getNetlinkFamily(isIPv6 bool) netlink.InetFamily {
 	if isIPv6 {
-		parameters = append(parameters, "-f", "ipv6")
+		return unix.AF_INET6
 	}
-	return parameters
+	return unix.AF_INET
 }
 
-// exec executes the conntrack tool using the given parameters
-func (ct *execCT) exec(parameters ...string) error {
-	conntrackPath, err := ct.execer.LookPath("conntrack")
-	if err != nil {
-		return fmt.Errorf("error looking for path of conntrack: %v", err)
+// getProtocolNumber return the Assigned Internet Protocol Number.
+// https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml
+func getProtocolNumber(proto v1.Protocol) uint8 {
+	switch proto {
+	case v1.ProtocolTCP:
+		return unix.IPPROTO_TCP
+	case v1.ProtocolUDP:
+		return unix.IPPROTO_UDP
+	case v1.ProtocolSCTP:
+		return unix.IPPROTO_SCTP
 	}
-	klog.V(4).InfoS("Clearing conntrack entries", "parameters", parameters)
-	output, err := ct.execer.Command(conntrackPath, parameters...).CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("conntrack command returned: %q, error message: %s", string(output), err)
-	}
-	klog.V(4).InfoS("Conntrack entries deleted", "output", string(output))
-	return nil
+	return 0
 }
 
-// ClearEntriesForIP is part of Interface
-func (ct *execCT) ClearEntriesForIP(ip string, protocol v1.Protocol) error {
-	parameters := parametersWithFamily(utilnet.IsIPv6String(ip), "-D", "--orig-dst", ip, "-p", protoStr(protocol))
-	err := ct.exec(parameters...)
-	if err != nil && !strings.Contains(err.Error(), noConnectionToDelete) {
+// ClearEntriesForIP delete the conntrack entries for connections specified by the
+// destination IP(original direction).
+func (ct *conntracker) ClearEntriesForIP(ip string, protocol v1.Protocol) error {
+	filter := &conntrackFilter{
+		protocol: getProtocolNumber(protocol),
+		original: &connectionTuple{
+			dstIP: netutils.ParseIPSloppy(ip),
+		},
+	}
+	klog.V(4).InfoS("Clearing conntrack entries", "org-dst", ip, "protocol", protocol)
+
+	n, err := ct.handler.ConntrackDeleteFilters(netlink.ConntrackTable, getNetlinkFamily(netutils.IsIPv6String(ip)), filter)
+	if err != nil {
 		// TODO: Better handling for deletion failure. When failure occur, stale udp connection may not get flushed.
 		// These stale udp connection will keep black hole traffic. Making this a best effort operation for now, since it
 		// is expensive to baby-sit all udp connections to kubernetes services.
-		return fmt.Errorf("error deleting connection tracking state for UDP service IP: %s, error: %v", ip, err)
+		return fmt.Errorf("error deleting connection tracking state for %s service IP: %s, error: %w", protoStr(protocol), ip, err)
```

Comment:
`protoStr` is just to convert the all-uppercase k8s API form to the all-lowercase `conntrack` CLI form. You should be able to remove it entirely as part of this PR. (And definitely just use `protocol` in log/error messages.)

Reply:
`protoStr` is dropped in the subsequent commit in this PR itself, 
`proxy/conntrack: consolidate flow cleanup` commit drops it. 

Should I drop `protoStr` in the `remove the conntrack binary dependency` commit itself?